### ISSUE TITLE
bsputil doesn't check for invalid arguments

### ISSUE THREAD

**Issue by ItEndsWithTens** [ericwa/ericw-tools#224]
Is it by design that bsputil swallows provided command line arguments if they're invalid? I was just trying to extract some textures from a map and had some trouble. After about half an hour of making sure I had the latest version, resetting file and directory permissions, and ultimately checking out the repo and stepping through the code in a debugger, I came to realize I'd been passing in --extract_textures instead of --extract-textures. I also remembered I've done that before. 😳

The code looks at the final argument first, and spits out the basic map info, like filename and BSP version number, then loops through the remaining args, but if they're not in the expected list of options it just ends the loop and proceeds to print a line of hyphens before exiting. The effect, as I saw, is a nicely formatted block of information about the level, but no output WAD or error message.

I'm certainly not looking to blame your software for my Big Dumb Idiot moment, but a warning that the provided argument was unexpected would have helped me understand my mistake.

**Comment by ericwa** [comment]
Oh no, sorry about that!
Yeah, it should abort with an error if any command line arguments are unrecognized.

I should probably also standardize the conventions (--xyz vs. -xyz, underscores vs dashes)  between the tools, keeping the old thing for compatibility.

**Comment by ericwa** [comment]
Fixed in 6c9962eb8c7e68659f8463ae29009f10aa64e452

It'll now print:
```
bsputil.exe --extract_entities C:\Users\Eric\Downloads\gmsp3\gmsp3v2.bsp
---- bsputil / ericw-tools 2.0.0-alpha8-17-g2354209f-dirty ----
************ ERROR ************
unknown option "extract_entities"

Process finished with exit code 1
```
